source/DataElement/AMDSFlatArray.cpp

```diff
@@ -0,0 +1,199 @@
+#include "source/DataElement/AMDSFlatArray.h"
+
+AMDSFlatArray::AMDSFlatArray(AMDSDataTypeDefinitions::DataType dataType, quint32 size)
+{
+	dataType_ = dataType;
+	resizeType(dataType_, size);
+}
+
+AMDSFlatArray::~AMDSFlatArray()
+{
+}
+
+AMDSFlatArray AMDSFlatArray::operator +(AMDSFlatArray &dataFlatArray)
+{
+	AMDSFlatArray newDataFlatArray(dataType(), size());
+	copyDataToTargetArray(&newDataFlatArray);
+
+	switch(dataType_) {
+	case AMDSDataTypeDefinitions::Signed8: {
+		QVector<qint8> *qint8SourceDataVector = &dataFlatArray.vectorQint8();
+		QVector<qint8> *qint8TargetDataVector = &newDataFlatArray.vectorQint8();
+		for (int i = 0; i < qint8TargetDataVector->size(); i++) {
```

Comment:
for loops like this should probably be of the form `for (int i = 0, size = qint8TartgetDataVector->size(); i < size; i++)` to save having to recompute the size on large arrays on each iteration.